src/redux/search/actionCreater.js

```diff
@@ -0,0 +1,49 @@
+import { API_KEY, API_URL } from 'react-native-dotenv';
+
+export const SEARCH_START = 'search_start';
+export const SEARCH_SUCCESS = 'search_success';
+export const SEARCH_FAILED = 'search_failed';
+
+export function searchStart() {
+  return {
+    type: SEARCH_START,
+  };
+}
+
+export function searchSuccess(result) {
+  return {
+    type: SEARCH_SUCCESS,
+    payload: result,
+  };
+}
+
+export function searchFailed(error) {
+  return {
+    type: SEARCH_FAILED,
+    payload: error,
+  };
+}
+
+export function searchData(input) {
+  return (dispatch) => {
+    const url = new URL(`${API_URL}/search/${input}`);
+    url.searchParams.append('token', API_KEY);
+    dispatch(searchStart());
+    fetch(url)
+      .then((response) => {
+        if (!response.ok) {
+          switch (response.status) {
+            case 404:
+              throw new Error('Resource not found or unknown symbols were provided.');
+            case 400:
+              throw new Error(`Invalid values were supplied when search for ${input}.`);
```

Comment:
Is this user error? Can they do something with this? Or does it need to involve a developer?